src/client/content_cao.cpp

```diff
@@ -1520,6 +1520,9 @@ void GenericCAO::processMessage(const std::string &data)
 
 		m_hp = result_hp;
 
+		if (m_is_local_player)
+			m_env->getLocalPlayer()->hp = m_hp;
```

Comment:
This line is a weird fix/workaround and shows that `LocalPlayer::hp` should be replaced directly with `LocalPlayer::parent->m_hp`.
I'd prefer to remove the changes here, instead fix it in another PR.

Reply:
> I'd prefer to remove the changes here, instead fix it in another PR.

If this part is removed, I'll have to remove `LocalPlayer::hp` in this PR itself. Otherwise, this fix wouldn't work. Is that what you requested?

Reply:
Removing this duplicated variable would be great, to get rid of out-of-sync problems. Doing that in a future PR is OK. I couldn't test this PR yet to check why this particular code here is even needed, hence I thought it was a fix for another issue.
Leave it as-is.

Reply:
This bit of code checks if the activeobject corresponds to the local player, and updates `LocalPlayer::hp` (which is used to check if player is dead) if so. I'll remove this bit in another PR as you suggest.